lightning/src/onion_message/messenger.rs

```diff
@@ -246,6 +246,20 @@ pub trait CustomOnionMessageHandler {
 	fn read_custom_message<R: io::Read>(&self, message_type: u64, buffer: &mut R) -> Result<Option<Self::CustomMessage>, msgs::DecodeError>;
 }
 
+/// An processed incoming onion message, containing either a Forward (another onion)
+/// or a Receive payload with decrypted contents
```

Comment:
nit:
```suggestion
/// or a Receive payload with decrypted contents.
```
and elsewhere